scripts/config.py

```diff
@@ -515,29 +521,30 @@ def _config_CUDA(self):
                 # in case user specified some wrapper of the compiler
                 # get rid of it: 'ccache g++' -> 'g++'
                 currentCxx = self.get(Config.KEY_CXX).split()[-1]
-                cxxVersion, cxxStrVersion = self._get_GCC_version(currentCxx)
                 nvccVersion, nvccFullVersion = self._get_CUDA_version(
                     self.configDict["NVCC"])
                 if self.configDict["CXX_CUDA"] == '':
                     print(
                         yellow('Checking for compatible GCC to be used with your CUDA'))
                     candidates = self._get_compatible_GCC(nvccVersion)
                     for c in candidates:
-                        p = 'g++-' + c
-                        if checkProgram(p, False):
-                            self.configDict["CXX_CUDA"] = p
+                        if checkProgram('g++-' + c, False):
+                            self.configDict["CXX_CUDA"] = 'g++-' + c
                             break
+                    if self.configDict["CXX_CUDA"] == '' and checkProgram('g++', False): #searching a g++ out of /usr/bin/...
```

Comment:
Like this you might set some g++, but you're not checking if it's compatible with CUDA.
So it seems to me that this code should be refactored a bit.
We should:
1. check if user set CXX_CUDA already
2. if not, check if CXX is compatible
3. if not, check if there is any other compatible compiler
4. ask user if they want to set the path manually (and check that the compiler is compatible)
5. if not, report warning and disable CUDA compilation

Reply:
I 've change the structure of the CXX_CUDA loops:
```
CXX_CUDA = os.environ.get("CXX_CUDA", "") #read env. var
if CXX_CUDA = ''
    search compiler candidates
    if not, search for compiler  outside /usr/bin/..
    ask the user for the compiler anyway
    check if the assigned compiler exists
    check if the allocated compiler is nvcc compatible

elif CXX_CUDA != ''
    check if the assigned compiler exists
    check if the allocated compiler is nvcc compatible

```